pkg/backend/cloud/client/client.go

```diff
@@ -146,6 +146,22 @@ func (pc *Client) DownloadTemplate(ctx context.Context, name string) (io.ReadClo
 	return resp.Body, resp.ContentLength, nil
 }
 
+// CheckIfCLINeedsUpdate asks the service if a newer version of the CLI is available and if we should issue a warning
+// or not.
+func (pc *Client) CheckIfCLINeedsUpdate(ctx context.Context, curVersion fmt.Stringer) (bool, string, error) {
+	reqQuery := apitype.CLIUpdateCheckRequest{
+		CurrentVersion: curVersion.String(),
+	}
+
+	var versionInfo apitype.CLIUpdateCheckResponse
+
+	if err := pc.restCall(ctx, "GET", "/api/cli/needs-update", reqQuery, nil, &versionInfo); err != nil {
+		return false, "", err
```

Comment:
Should we ignore errors here? The idea is to just give the user a prod to upgrade, so if we miss one here, it's not the end of the world ... My only concern would be if things get permanently busted, it may go unnoticed, but for that we should just have a targeted unit test.

Reply:
When [we call this method](https://github.com/pulumi/pulumi/pull/1696/files#diff-12ac49da96dc03d6252376229cc3afccR86) if an error was returned, we simply ignore it without failing.  I think making that decision at the call-site instead of the client reads better overall?

Reply:
Agree 100% that this is the right factoring of concerns. Sorry I missed the callsite!